cmd/kubeadm/app/preflight/checks_test.go

```diff
@@ -556,3 +557,57 @@ func restoreEnv(e map[string]string) {
 		os.Setenv(k, v)
 	}
 }
+
+func TestKubeletVersionCheck(t *testing.T) {
+	type T struct {
+		kubeletVersion string
+		k8sVersion     string
+		expectErrors   bool
+		expectWarnings bool
+	}
+
+	cases := []T{
+		{"v1.7.3", "v1.7.8", true, false},          // Too old kubelet
+		{"v1.9.0", "v1.9.5", false, false},         // kubelet within same major.minor as control plane
+		{"v1.9.0", "v1.10.1", false, false},        // kubelet is lower than control plane, but newer than minimally supported
+		{"v1.10.0-alpha.1", "v1.9.1", false, true}, // kubelet is newer than control plane
+		{"v1.10.2", "", false, false},              // check minimally supported version when there is no information about control plane
+	}
+
+	dir, err := ioutil.TempDir("", "test-kubelet-version-check")
+	if err != nil {
+		t.Errorf("Failed to create directory for testing GetKubeletVersion: %v", err)
+	}
+	defer os.RemoveAll(dir)
+
+	// We don't want to call real kubelet or something else in $PATH
+	oldPATH := os.Getenv("PATH")
```

Comment:
Can you please re-engineer this to use `k8s.io/utils/exec` for this instead so you can fake out the exec calls in unit tests instead of doing this?

Reply:
I haven't noticed that lib. Will check it. From quick look, it might require change also in implementation of `GetKubeletVersion()`. Do you want to have it as part of this PR or maybe let's do re-factoring to use utils/exec as separate PR ?

Reply:
Follow up is fine by me